packages/data/lib/database/dao/movie_dao.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:domain/entities/db/genre.dart';
+import 'package:domain/entities/db/movie.dart';
+import 'package:floor/floor.dart';
+
+@dao
+abstract class MovieDao {
+  @Query('SELECT * FROM Movie WHERE moviesType = :movieType')
+  Future<List<Movie>> findMoviesByType(int movieType);
+
+  @Query('DELETE FROM Movie WHERE id in (:ids)')
+  Future<void> deleteMoviesWithIds(List<int> ids);
+
+  @Insert(onConflict: OnConflictStrategy.replace)
+  Future<void> insertMovies(List<Movie> movies);
+
+  @insert
+  Future<void> insertGenres(List<Genre> genres);
```

Comment:
seems that this method should be in genre_dao

Reply:
For now only decision I found, later I'll try to pass full database object instead of dao as dependencies